crates/precompile/src/bls12_381/utils.rs

```diff
@@ -33,39 +19,8 @@ pub(super) fn remove_padding(input: &[u8]) -> Result<&[u8; FP_LENGTH], Precompil
     Ok(unpadded.try_into().unwrap())
 }
 
-/// Extracts a scalar from a 32 byte slice representation, decoding the input as a big endian
-/// unsigned integer. If the input is not exactly 32 bytes long, an error is returned.
-///
-/// From [EIP-2537](https://eips.ethereum.org/EIPS/eip-2537):
-/// * A scalar for the multiplication operation is encoded as 32 bytes by performing BigEndian
-///   encoding of the corresponding (unsigned) integer.
-///
-/// We do not check that the scalar is a canonical Fr element, because the EIP specifies:
-/// * The corresponding integer is not required to be less than or equal than main subgroup order
-///   `q`.
-pub(super) fn extract_scalar_input(input: &[u8]) -> Result<blst_scalar, PrecompileError> {
-    if input.len() != SCALAR_LENGTH {
-        return Err(PrecompileError::Other(format!(
-            "Input should be {SCALAR_LENGTH} bytes, was {}",
-            input.len()
-        )));
-    }
-
-    let mut out = blst_scalar::default();
-    // SAFETY: `input` length is checked previously, out is a blst value.
-    unsafe {
-        // Note: We do not use `blst_scalar_fr_check` here because, from EIP-2537:
-        //
-        // * The corresponding integer is not required to be less than or equal than main subgroup
-        // order `q`.
-        blst_scalar_from_bendian(&mut out, input.as_ptr())
-    };
-
-    Ok(out)
-}
-
 /// Checks if the input is a valid big-endian representation of a field element.
-fn is_valid_be(input: &[u8; 48]) -> bool {
+pub(super) fn is_valid_be(input: &[u8; 48]) -> bool {
     for (i, modulo) in input.iter().zip(MODULUS_REPR.iter()) {
```

Comment:
unrelated but arrays/slices can be compared directly: `*input < MODULUS_REPR`

Reply:
That's a good point, let me change it